CompanyEmployees.Infrastructure.Persistence/Repositories/AttendanceRepository.cs

```diff
@@ -0,0 +1,41 @@
+using CompanyEmployees.Core.Domain.Entities;
+using CompanyEmployees.Core.Domain.Repositories;
+using Microsoft.EntityFrameworkCore;
+
+namespace CompanyEmployees.Infrastructure.Persistence.Repositories;
+
+internal sealed class AttendanceRepository : RepositoryBase<Attendance>, IAttendanceRepository
+{
+    public AttendanceRepository(RepositoryContext repositoryContext)
+        : base(repositoryContext)
+    {
+    }
+
+    public async Task<IEnumerable<Attendance>> GetAllAttendancesAsync(bool trackChanges, CancellationToken ct = default) =>
+        await FindAll(trackChanges)
+            .OrderByDescending(a => a.CheckInTime)
+            .ToListAsync(ct);
+
```

Comment:
Consider marking `GetAttendanceAsync` as returning a nullable `Attendance?` to clearly indicate that null may be returned when no matching record is found. Other methods like `GetTodayAttendanceByEmployeeIdAsync` already return nullable entities, and it would be best to be consistent and explicit.

```suggestion
public async Task<Attendance?> GetAttendanceAsync(Guid attendanceId, bool trackChanges, CancellationToken ct = default) =>
```